packages/taco-auth/src/providers/eip4361/external-eip4361.ts

```diff
@@ -1,18 +1,62 @@
 import { SiweMessage } from 'siwe';
 
+import { AuthProvider } from '../../auth-provider';
+
 import { EIP4361_AUTH_METHOD, EIP4361AuthSignature } from './auth';
 
-export const USER_ADDRESS_PARAM_EXTERNAL_EIP4361 =
-  ':userAddressExternalEIP4361';
+async function generateAndVerifySiweMessage(
+  message: string,
+  signature: string,
+): Promise<SiweMessage> {
+  const siweMessage = new SiweMessage(message);
+  // this will trigger validation for the signature and all parameters (`expirationTime`, `notBefore`...).
+  await siweMessage.verify({ signature });
+  const twoHoursBeforeNow = new Date(Date.now() - 2 * 60 * 60 * 1000);
+  if (!siweMessage.issuedAt) {
+    throw new Error(
+      `The message does not have an \`issuedAt\` field. This case should never happen and it would cause the message to be rejected by TACo nodes.`,
```

Comment:
the "this should never happen" is really an internal dev note.

```suggestion
      // this should never happen
      `The SIWE message is missing an \`issuedAt\` field and would be rejected by TACo nodes.`,
```